### PR TITLE
ABIOS 3097 Swift 4 migration

### PR DESCRIPTION
**Pull request by facumenzella**
Swift 4 migration.

bitmoji

### CODE REVIEW

File: bumper/BumperCell.swift [hunk 1]
```diff
@@ -37,9 +37,9 @@ class BumperCell: UITableViewCell {
         views["titleLabel"] = titleLabel
         views["infoLabel"] = infoLabel
 
-        infoLabel.setContentCompressionResistancePriority(UILayoutPriorityRequired, for: .horizontal)
-        infoLabel.setContentHuggingPriority(UILayoutPriorityRequired, for: .horizontal)
-        titleLabel.setContentCompressionResistancePriority(UILayoutPriorityRequired, for: .vertical)
+        infoLabel.setContentCompressionResistancePriority(UILayoutPriority.required, for: .horizontal)
+        infoLabel.setContentHuggingPriority(UILayoutPriority.required, for: .horizontal)
+        titleLabel.setContentCompressionResistancePriority(UILayoutPriority.required, for: .vertical)
```
Comment (ahernandezlopez) [inline bumper/BumperCell.swift]: we can just leave `.required`

File: bumper/BumperViewController.swift [hunk 2]
```diff
@@ -107,11 +107,11 @@ fileprivate extension BumperViewController {
         enableBumperSwitch.addTarget(self, action: #selector(switchValueChanged), for: .valueChanged)
     }
 
-    dynamic func switchValueChanged() {
+    @objc dynamic func switchValueChanged() {
         viewModel.setEnabled(enableBumperSwitch.isOn)
     }
 
-    dynamic func dismissViewController() {
+    @objc dynamic func dismissViewController() {
```
Comment (ahernandezlopez) [inline bumper/BumperViewController.swift]: `dynamic` keyword can be removed

File: bumper/BumperViewController.swift [hunk 3]
```diff
@@ -107,11 +107,11 @@ fileprivate extension BumperViewController {
         enableBumperSwitch.addTarget(self, action: #selector(switchValueChanged), for: .valueChanged)
     }
 
-    dynamic func switchValueChanged() {
+    @objc dynamic func switchValueChanged() {
```
Comment (ahernandezlopez) [inline bumper/BumperViewController.swift]: `dynamic` keyword can be removed

File: bumper/BumperDAO.swift [hunk 4]
```diff
@@ -9,8 +9,8 @@
 import Foundation
 
 protocol BumperDAO {
-    func setBool(_ value: Bool, forKey defaultName: String)
-    func boolForKey(_ defaultName: String) -> Bool
-    func stringForKey(_ defaultName: String) -> String?
-    func setObject(_ value: Any?, forKey defaultName: String)
+    func set(_ value: Bool, forKey defaultName: String)
```
Comment (juanif) [inline bumper/BumperDAO.swift]: do we want to keep the _ on method definitions?? do we have any "rule" for this @letgoapp/ios ?

File: bumper/BumperCell.swift (hunk 1 above)
Comment (facumenzella) [reply bumper/BumperCell.swift]: Didn't even noticed it.

On it!!

File: bumper/BumperDAO.swift (hunk 4 above)
Comment (ahernandezlopez) [reply bumper/BumperDAO.swift]: We don't have a rule nope, but will be nice to quickly talk about it. This mimics the definition of `UserDefaults` methods right?

Comment (facumenzella) [reply bumper/BumperDAO.swift]: Exactly.

**Pull request merged**